src/imp_pl.rs

```diff
@@ -56,15 +58,22 @@ impl<T> OnceCell<T> {
             //   but that is more complicated
             // - finally, if it returns Ok, we store the value and store the flag with
             //   `Release`, which synchronizes with `Acquire`s.
-            let value = f()?;
-            // Safe b/c we have a unique access and no panic may happen
-            // until the cell is marked as initialized.
-            let slot: &mut Option<T> = unsafe { &mut *self.value.get() };
-            debug_assert!(slot.is_none());
-            *slot = Some(value);
-            self.is_initialized.store(true, Ordering::Release);
-        }
-        Ok(())
+            let f = f.take().unwrap_or_else(|| unsafe { hint::unreachable_unchecked() });
```

Comment:
This probably shaves off a couple of byets, right? Let's exctract this to  an `unwrap_unchecked` function into the parent module, and use it for `impl_std` as well.

Reply:
`unwrap_or_else` is short enough, I don't think it needs to be extracted into a separate function.

Reply:
fine by me, as the two code paths for pl and for std are identical